packages/core/src/store/__tests__/level-store.test.ts

```diff
@@ -0,0 +1,40 @@
+import { jest } from '@jest/globals'
+import tmp from 'tmp-promise'
+import { LevelDbStore } from '../level-db-store.js'
+import { TestUtils } from '@ceramicnetwork/common'
+
+describe("LevelStore", () => {
+  let tmpFolder: any
+  let levelStore: LevelDbStore
+
+  beforeEach(async () => {
+    tmpFolder = await tmp.dir({ unsafeCleanup: true })
+    levelStore = new LevelDbStore(tmpFolder.path, 'fakeNetwork')
+
+    // add a small delay after creating the leveldb instance before trying to use it.
+    await TestUtils.delay(100)
+  })
+
+  afterEach(async () => {
+    await levelStore.close()
+    await tmpFolder.cleanup()
+  })
+
+  test("The main store is separated from a sub-channel", async () => {
```

Comment:
I think this file should also test listing all keys in the stores and making sure that the list of keys is separate across the sub-channels as well.  That's where I'm most expecting a bug to show up (especially with the S3 store): `pin.ls()` just iterates over all the keys in the store